### DISCUSSION TITLE
Tileable open_simplex

### DISCUSSION THREAD

**Discussion by ftomassetti** [Razaekel/noise-rs#259] [Ideas]
I think it would be very useful to have a function or an example of how to use open_simplex to generate a tileable image. Possibly tileable on the horizontal, vertical or both sides

**Comment by Razaekel** [comment]
The method to create tilable procedural noise is to make each input dimension get mapped onto a circle, and then use the equivalent point on the circle to get determine the noise value. So 2d procedural noise would require 4d noise, and 3d noise would require 6d. Not that hard to figure out, just takes some time to implement. You could do it yourself by mapping the input values and then feeding them to the correct noise generator. Would only work with 2d since there's implementations for 3d and 4d. You could do it with 3d if you only wanted to tile it on one axis.

**Comment by real-felix** [comment]
Hi, is it possible to add an example about how to do so? I'm trying right now to create a wrapping heightmap by combining several `NoiseFn`s, but I don't know if I can keep this setup and add my own wrapping `NoiseFn`, or if I must reimplement all from scratch?

**Comment by real-felix** [comment]
Sorry about the double post, but I've found by myself how to do this. [Here is the code][1].

This is very easy to create a tileable 2D texture with your crate, and I'd like to make a PR to add this example, but I'm not sure what I should push. The whole color interpolation, while it render nicely, has nothing to do with noise generation.

[1]: https://gitlab.com/Boiethios/wrapping-heightmap/commit/a7566776c21c94a6f0f4905dc9286e891bdfcbe4#4b569f42a6967dec04275af54f4ca9ab6a4eee64
